tests/unit/check_cl_cdm.c

```diff
@@ -0,0 +1,141 @@
+ * limitations under the License.
+ */
+
+#define TEST_SUITE_NAME "check_cl_cdm"
+
+#include "mock_hostmod.h"
+#include "testutil.h"
+
+#include <osd/cl_cdm.h>
+#include <osd/osd.h>
+#include <osd/reg.h>
+
+struct osd_hostmod_ctx *hostmod_ctx;
+struct osd_log_ctx *log_ctx;
+
+// DI address of the CDM module to be tested; chosen arbitrarily
+const unsigned int cdm_diaddr = 14;
+
+/**
+ * Test fixture: setup (called before each tests)
+ */
+void setup(void) { mock_hostmod_setup(); }
+
+/**
+ * Test fixture: setup (called after each test)
+ */
+void teardown(void) { mock_hostmod_teardown(); }
+
+
+START_TEST(test_get_desc)
+{
+    osd_result rv;
+
+    mock_hostmod_expect_mod_describe(cdm_diaddr, OSD_MODULE_VENDOR_OSD,
+                                     OSD_MODULE_TYPE_STD_CDM, 0);
+
+    mock_hostmod_expect_reg_read16(1, cdm_diaddr, OSD_REG_CDM_CORE_CTRL,
+                                   OSD_OK);
+
+    mock_hostmod_expect_reg_read16(5, cdm_diaddr, OSD_REG_CDM_CORE_REG_UPPER,
+                                   OSD_OK);
+
+    mock_hostmod_expect_reg_read16(32, cdm_diaddr, OSD_REG_CDM_CORE_DATA_WIDTH,
+                                   OSD_OK);
+    
+    struct osd_cdm_desc cdm_desc;
+    rv = osd_cl_cdm_get_desc(mock_hostmod_get_ctx(), cdm_diaddr, &cdm_desc);
+    ck_assert_int_eq(rv, OSD_OK);
+    ck_assert_uint_eq(cdm_desc.core_ctrl, 1);
+    ck_assert_uint_eq(cdm_desc.core_reg_upper, 5);
+    ck_assert_uint_eq(cdm_desc.core_data_width, 32);
+    ck_assert_uint_eq(cdm_desc.di_addr, cdm_diaddr);
+}
+END_TEST
+
+START_TEST(test_get_desc_wrong_module)
+{
+    osd_result rv;
+
+    mock_hostmod_expect_mod_describe(cdm_diaddr, OSD_MODULE_VENDOR_OSD,
+                                     OSD_MODULE_TYPE_STD_CTM, 0);
+
+    struct osd_cdm_desc cdm_desc;
+    rv = osd_cl_cdm_get_desc(mock_hostmod_get_ctx(), cdm_diaddr, &cdm_desc);
+    ck_assert_int_eq(rv, OSD_ERROR_WRONG_MODULE);
+}
+END_TEST
+
+static void event_handler(void *arg, const struct osd_cdm_desc *cdm_desc,
+                          const struct osd_cdm_event *event)
+{
+    ck_assert(arg);
+    ck_assert(cdm_desc);
+    ck_assert(event);
+
+    // in this test we pass the expected event as callback argument
+    struct osd_cdm_event *exp_event = arg;
+
+    ck_assert_uint_eq(exp_event->stall, event->stall);
+    
+}
+
+START_TEST(test_handle_event)
+{
+    osd_result rv;
+
+    struct osd_cdm_desc cdm_desc;
+    cdm_desc.di_addr = 2;
+    cdm_desc.core_ctrl = 16;
+    cdm_desc.core_reg_upper = 1;
+    cdm_desc.core_data_width = 32;
+
+
+    struct osd_cdm_event_handler ev_handler;
+    ev_handler.cb_fn = event_handler;
+    ev_handler.cdm_desc = &cdm_desc;
+
+    bool stall_bit = 1;
+    struct osd_packet *pkg_trace;
+    osd_packet_new(&pkg_trace, osd_packet_sizeconv_payload2data(1));
+    rv = osd_packet_set_header(pkg_trace, 1, 2, OSD_PACKET_TYPE_EVENT, 0);
+    ck_assert_int_eq(rv, OSD_OK);
+    pkg_trace->data.payload[0] = stall_bit; // stall
```

Comment:
What you actually try to do here is assigning a single bit to bit position 0. To make this operation clear you can write it as (1 << 0). Even more readable is using a define for "0" (you can find examples reg.h) and then making use of the BIT() macro defined in osd.h.